ansible/playbooks/roles/repository/files/download-requirements/download-requirements.py

```diff
@@ -32,31 +32,27 @@ def rerun_download_requirements(config: Config):
     """
     additional_args: List[str] = ['--rerun']
 
-    # carry over info about installed 3rd party tools and modules:
-    if config.pip_installed:
-        additional_args.append('--pip-installed')
+    # carry over info about installed 3rd party modules:
+    if config.pyyaml_installed:
+        additional_args.append('--pyyaml-installed')
 
-    if config.poyo_installed:
-        additional_args.append('--poyo-installed')
+    cmd: List[str] = [sys.executable, __file__] + sys.argv[1:] + additional_args
```

Comment:
Why using sys.executable when sys.argv[0] already contains that information? In line 43 we pass this variable twice, why?

Reply:
sys.executable contains path to python interpreter, e.g. `/usr/libexec/platform-python` when the script is run as:
`/usr/libexec/platform-python download-requirements.py`

sys.argv[0] contains only the script name.

sys.executable is used twice by purpose to have the same value in CMD column of `ps` command (`/proc/<PID>/cmdline`) after rerun the script.

More info here: https://stackoverflow.com/questions/31447442/difference-between-os-execl-and-os-execv-in-python